src/pj_param.c

```diff
@@ -1,20 +1,78 @@
 /* put parameters in linked list and retrieve */
-#include <projects.h>
 #include <stdio.h>
+#include <ctype.h>
 #include <string.h>
+#include "projects.h"
 
 /* create parameter list entry */
 paralist *pj_mkparam(char *str) {
-	paralist *newitem;
-
-	if((newitem = (paralist *)pj_malloc(sizeof(paralist) + strlen(str))) != NULL) {
-		newitem->used = 0;
-		newitem->next = 0;
-		if (*str == '+')
-			++str;
-		(void)strcpy(newitem->param, str);
-	}
-	return newitem;
+    paralist *newitem;
+
+    if((newitem = (paralist *)pj_malloc(sizeof(paralist) + strlen(str))) != NULL) {
+        newitem->used = 0;
+        newitem->next = 0;
+        if (*str == '+')
+            ++str;
+        (void)strcpy(newitem->param, str);
+    }
+    return newitem;
+}
+
+
+/* As pj_mkparam, but payload is terminated by whitespace, rather than 0 */
```

Comment:
`0` -> `\0` or `NUL`

Reply:
NUL does not exist in C, \0 should be '\0', but really just indicates an octal character constant (which in context is promoted to an int) - and that is expressed with less clutter as a plain decimal integer constant.

Reply:
That is just slang, `\0` is called [NUL](https://en.wikipedia.org/wiki/Null_character) in ASCII. People don't talk about 0-terminated strings, but [null-terminated strings](https://en.wikipedia.org/wiki/Null-terminated_string).

Reply:
(or "strings in ASCIIZ format", if they're old enough)

Reply:
Since this is a comment, it's not super critical.  It was just that someone could (when overtired) interpret `0` as `'0'` (`\x30`).  NUL and \0 in a comment should be less likely to be read incorrectly by a human.

Reply:
Changed to *As pj_mkparam, but payload ends at first whitespace, rather than at end of **str***